packages/block-library/src/site-logo/index.php

```diff
@@ -71,56 +71,83 @@ function register_block_core_site_logo() {
 			'render_callback' => 'render_block_core_site_logo',
 		)
 	);
-	add_filter( 'pre_set_theme_mod_custom_logo', 'sync_site_logo_to_theme_mod' );
-	add_filter( 'theme_mod_custom_logo', 'override_custom_logo_theme_mod' );
 }
 add_action( 'init', 'register_block_core_site_logo' );
 
 /**
- * Overrides the custom logo with a site logo, if the option is set.
- *
- * @param string $custom_logo The custom logo set by a theme.
- *
- * @return string The site logo if set.
+ * Expose the custom_logo theme-mod in the settings REST API.
  */
-function override_custom_logo_theme_mod( $custom_logo ) {
-	$site_logo = get_option( 'site_logo' );
-	return false === $site_logo ? $custom_logo : $site_logo;
-}
+register_setting(
+	'general',
+	'theme_mods_' . get_option( 'stylesheet' ),
+	array(
+		'type'         => 'object',
+		'show_in_rest' => array(
+			'name'   => 'theme_mods_' . get_option( 'stylesheet' ),
+			'schema' => array(
+				'type'       => 'object',
+				'properties' => array(
+					'custom_logo' => array( 'type' => 'integer' ),
+				),
+			),
+		),
+	)
+);
 
 /**
- * Syncs the site logo with the theme modified logo.
+ * Expose the "stylesheet" setting in the REST API.
+ */
+register_setting(
```

Comment:
The `stylesheet` setting is never properly registered in core, so we need to register it here and expose it to the REST API.